src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/NrbfDecoder.cs

```diff
@@ -69,28 +69,21 @@ public static bool StartsWithPayloadHeader(Stream stream)
             return false;
         }
 
-        try
+        byte[] buffer = new byte[SerializedStreamHeaderRecord.Size];
+        int offset = 0;
+        while (offset < buffer.Length)
         {
-#if NET
-            Span<byte> buffer = stackalloc byte[SerializedStreamHeaderRecord.Size];
-            stream.ReadExactly(buffer);
-#else
-            byte[] buffer = new byte[SerializedStreamHeaderRecord.Size];
-            int offset = 0;
-            while (offset < buffer.Length)
+            int read = stream.Read(buffer, offset, buffer.Length - offset);
+            if (read == 0)
             {
-                int read = stream.Read(buffer, offset, buffer.Length - offset);
-                if (read == 0)
-                    throw new EndOfStreamException();
-                offset += read;
+                return false;
```

Comment:
the goal of this change is to simply return `false` rather than throw `EOSE`

Reply:
Do we need to set the stream position back to the beginning before we return here?

Reply:
> Do we need to set the stream position back to the beginning before we return here?

Great catch!